src/ovirtapi/index.js

```diff
@@ -131,24 +131,22 @@ const OvirtApi = {
     }
     return httpGet({ url })
   },
-  getVmsByPage ({ page, additional }: { page: number, additional: Array<string> }): Promise<Object> {
-    assertLogin({ methodName: 'getVmsByPage' })
-    let url = `${AppConfiguration.applicationContext}/api/vms/;max=${AppConfiguration.pageLimit}?search=SORTBY NAME ASC page ${page}`
-    if (additional && additional.length > 0) {
-      url += `&follow=${additional.join(',')}`
-    }
-    return httpGet({ url })
-  },
-  getVmsByCount ({ count, additional }: { count: number, additional: Array<string> }): Promise<Object> {
-    assertLogin({ methodName: 'getVmsByCount' })
-    let url = `${AppConfiguration.applicationContext}/api/vms/;max=${count}?search=SORTBY NAME ASC`
-    if (additional && additional.length > 0) {
-      url += `&follow=${additional.join(',')}`
-    }
+  getVms ({ count, page, additional }: { count?: number, page?: number, additional?: Array<string> }): Promise<Object> {
+    assertLogin({ methodName: 'getAllVms' })
+    const max = count ? `;max=${count}` : ''
+    const params =
+      [
+        page ? 'search=' + encodeURIComponent(`SORTBY NAME ASC page ${page}`) : '',
+        additional && additional.length > 0 ? 'follow=' + encodeURIComponent(`${additional.join(',')}`) : '',
+      ]
+        .filter(p => !!p)
+        .join('&')
+
+    const url = `${AppConfiguration.applicationContext}/api/vms${max}${params ? '?' : ''}${params}`
```

Comment:
consider extracting URL building to a method that could be used by both vms and pools

Reply:
It probably the only place where the URL building code matches that much.  I'd much rather give the entire API file a refactor to address things like this.